platform/pwl.c

```diff
@@ -816,68 +824,68 @@ window_dispatch_key_event (PwlWindow  *self,
 
 
 static gboolean
-display_on_keyboard_repeat_timeout (PwlDisplay *self)
+display_on_keyboard_repeat_timeout (PwlSeat *pwl_seat)
 {
-    if (!self->keyboard_target) {
+    if (!pwl_seat->keyboard_target) {
         g_critical ("%s: No keyboard event target window!", G_STRFUNC);
         /*
          * We cannot use display_stop_keyboard_repeat() here because it
          * would destroy the GSource while it is still being dispatched.
          */
-        g_clear_pointer (&self->keyboard_repeat_source, g_source_unref);
+        g_clear_pointer (&pwl_seat->keyboard_repeat_source, g_source_unref);
         return G_SOURCE_REMOVE;
     }
 
-    window_dispatch_key_event (self->keyboard_target,
-                               &self->xkb_data,
-                               self->keyboard_repeat_timestamp,
-                               self->keyboard_repeat_key,
-                               self->keyboard_repeat_state);
+    window_dispatch_key_event (pwl_seat->keyboard_target,
+                               &pwl_seat->xkb_data,
+                               pwl_seat->keyboard_repeat_timestamp,
+                               pwl_seat->keyboard_repeat_key,
+                               pwl_seat->keyboard_repeat_state);
     return G_SOURCE_CONTINUE;
 }
 
 
 static void
-display_stop_keyboard_repeat (PwlDisplay *self)
+display_stop_keyboard_repeat (PwlSeat *pwl_seat)
 {
-    if (!self->keyboard_repeat_source)
+    if (!pwl_seat->keyboard_repeat_source)
         return;
 
-    g_source_destroy (self->keyboard_repeat_source);
-    g_clear_pointer (&self->keyboard_repeat_source, g_source_unref);
+    g_source_destroy (pwl_seat->keyboard_repeat_source);
+    g_clear_pointer (&pwl_seat->keyboard_repeat_source, g_source_unref);
 }
 
 
 static void
-display_start_keyboard_repeat (PwlDisplay *self,
+display_start_keyboard_repeat (PwlSeat    *pwl_seat,
```

Comment:
Same here, I would rename the function to:

```c
static void
seat_start_keyboard_repeat (PwlSeat *self,
                            /* the rest stays the same... */
```

Reply:
done.